src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -50,6 +51,33 @@ FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::~FFN()
       boost::apply_visitor(deleteVisitor));
 }
 
+template<typename OutputLayerType, typename InitializationRuleType,
+         typename... CustomLayers>
+void FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::CheckInputShape(
+    size_t inputShape, std::string functionName)
+{
+  for (size_t l=0; l<network.size(); ++l)
+  {
+    size_t layerInShape = boost::apply_visitor(InShapeVisitor(), network[l]);
+    if (layerInShape == 0)
+    {
+      continue;
+    }
+    else if (layerInShape == inputShape)
+    {
+      break;
+    }
+    else
+    {
+      std::string estr = "FFN<>::" + functionName + ": "; 
+                  estr += "the first layer of the network expects ";
+                  estr += std::to_string(layerInShape) + " elements, ";
+                  estr += "but the input has " + std::to_string(inputShape) + " rows! ";
+      throw std::logic_error(estr);
```

Comment:
Not sure, if the function name is necessary, if we apply the same functionality to the GAN and RNN class, I guess it would make it easier if we don't have the function name here.

Reply:
I added function name because the user might use both `Train()` and `Predict()` in the same program and function name might help in debugging and informing the name of function where the user gave wrong input.